### PR TITLE
DWR-157

### PR DESCRIPTION
**Pull request by malaffoon**
This implementation follows what is outlined in DWR-157 for replaying imports that weren't initially processed. It does not deal with replaying imports that were initially processed through the system and need to be reprocessed for some reason.

### CODE REVIEW

File: common/src/main/java/org/opentestsystem/rdw/ingest/common/model/ImportStatus.java [hunk 1]
```diff
@@ -16,8 +16,8 @@
     BAD_FORMAT(-2),
     INVALID(-1),
     ACCEPTED(0),
-    PROCESSED(1),
-    PUBLISHED(2);
```
Comment (malaffoon) [inline common/src/main/java/org/opentestsystem/rdw/ingest/common/model/ImportStatus.java]: Unrelated cleanup ... we aren't using the PUBLISHED state in the data warehouse.

File: common/src/main/java/org/opentestsystem/rdw/ingest/common/model/RdwImport.java [hunk 2]
```diff
@@ -14,8 +14,10 @@
     private String digest;
     private ImportStatus status;
     private String batch;
+    private Long replayId;
```
Comment (malaffoon) [inline common/src/main/java/org/opentestsystem/rdw/ingest/common/model/RdwImport.java]: This is part of this change; the column is used to keep track of replay attempts.

File: common/src/main/java/org/opentestsystem/rdw/ingest/common/model/RdwImport.java [hunk 3]
```diff
@@ -14,8 +14,10 @@
...
     private String creator;
     private Instant created;
+    private Instant updated;
```
Comment (malaffoon) [inline common/src/main/java/org/opentestsystem/rdw/ingest/common/model/RdwImport.java]: This is unrelated, just a column we added to the import table.

File: import-service/src/main/java/org/opentestsystem/rdw/ingest/service/DefaultImportService.java [hunk 4]
```diff
@@ -95,25 +98,85 @@ public RdwImport importContent(final SbacUserDetails user,
...
+        if (reserved == 0) {
+            return 0;
         }
+
+        final String batch = "BATCH" + Long.toString(batchId);
+        long count = 0;
+        for (final RdwImport rdwImport : repository.findByReplayId(batchId)) {
+            try {
+                // get the payload
+                final String location = location(rdwImport);
```
Comment (malaffoon) [inline import-service/src/main/java/org/opentestsystem/rdw/ingest/service/DefaultImportService.java]: There is some code duplication, but when i started to factor it out, it made the overall code harder to read. I might take another shot at it.

File: import-service/src/main/java/org/opentestsystem/rdw/ingest/service/DefaultImportService.java [hunk 5]
```diff
@@ -95,25 +98,85 @@ public RdwImport importContent(final SbacUserDetails user,
     }
 
     @Override
-    public long resubmitImports(RdwImportQuery query) {
-        // if there is no query, resort to some reasonable default
-        if (query == null || query.isEmpty()) {
```
Comment (malaffoon) [inline import-service/src/main/java/org/opentestsystem/rdw/ingest/service/DefaultImportService.java]: This code was never really being invoked (except in tests) since the controller was also setting up a "reasonable default".

File: import-service/src/main/resources/application.sql.yml [hunk 6]
```diff
@@ -17,18 +17,39 @@ sql:
       SELECT * FROM import WHERE id=:id
 
     findOneByDigest: >-
-      SELECT * FROM import WHERE digest=:digest
+      SELECT * FROM import WHERE digest=:digest AND replay_id IS NULL
```
Comment (malaffoon) [inline import-service/src/main/resources/application.sql.yml]: i verified that this query is using the index on `digest` even with the extra condition

File: import-service/src/test/java/org/opentestsystem/rdw/ingest/service/DefaultImportServiceTest.java [hunk 7]
```diff
@@ -12,18 +11,20 @@
 import org.opentestsystem.rdw.ingest.common.model.RdwImport;
 import org.opentestsystem.rdw.ingest.common.model.RdwImportQuery;
 import org.opentestsystem.rdw.ingest.repository.RdwImportRepository;
-import org.opentestsystem.rdw.utils.TenancyChain;
 
 import java.util.List;
 import java.util.Properties;
+import java.util.Random;
```
Comment (malaffoon) [inline import-service/src/test/java/org/opentestsystem/rdw/ingest/service/DefaultImportServiceTest.java]: oops, this isn't being used ... will remove.

File: migrate-reporting/src/main/resources/application.yml [hunk 8]
```diff
@@ -9,7 +9,7 @@ server:
 migrate:
   batch:
     size: 1000
-    delay: 300000
+    delay: 60000
```
Comment (malaffoon) [inline migrate-reporting/src/main/resources/application.yml]: since i keep changing this to 1 minute, i figure i'll check it in this time

File: common/src/main/java/org/opentestsystem/rdw/ingest/common/model/RdwImportQuery.java [hunk 9]
```diff
@@ -130,8 +141,10 @@ public Builder content(final ImportContent content) {
         }
 
         public Builder content(final String content) {
-            this.content = isNullOrEmpty(content) ? null : ImportContent.valueOf(content.toUpperCase());
-            return this;
+            if (isNullOrEmpty(content)) {
```
Comment (malaffoon) [inline common/src/main/java/org/opentestsystem/rdw/ingest/common/model/RdwImportQuery.java]: A little refactoring just to make things more consistent/readable.

File: common/src/main/java/org/opentestsystem/rdw/ingest/common/model/RdwImport.java [hunk 10]
```diff
@@ -86,28 +100,43 @@ public static Builder builder() {
...
             rdwImport.content = content;
             rdwImport.contentType = contentType;
             rdwImport.digest = digest;
             rdwImport.status = status;
             rdwImport.batch = batch;
+            rdwImport.replayId = replayId;
             rdwImport.creator = creator;
-            rdwImport.created = created;
+            rdwImport.created = created != null ? created : now();
+            rdwImport.updated = updated != null ? updated : rdwImport.created;
```
Comment (agorina) [inline common/src/main/java/org/opentestsystem/rdw/ingest/common/model/RdwImport.java]: Just curious: what are the benefits of setting it in Java vs relying on the mySQL. Just more control?

Comment (malaffoon) [inline common/src/main/java/org/opentestsystem/rdw/ingest/common/model/RdwImport.java]: None really, just following the pattern (yeah, i know i established the pattern).
